File: shared/src/commonMain/kotlin/nl/fbdevelopment/healthyplatform/data/models/dto/message/TimeLinePostDto.kt
@file:OptIn(ExperimentalUuidApi::class)

package nl.fbdevelopment.healthyplatform.data.models.dto.message

import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonObject
import nl.fbdevelopment.healthyplatform.data.models.dto.user.PublicUserDto
import kotlin.uuid.ExperimentalUuidApi
import kotlin.uuid.Uuid

@Serializable
data class TimeLinePostDto(
    val uuid: Uuid,
    val createdAt: Long,
    val updatedAt: Long,
    val deletedAt: Long?,
    val content: JsonObject,
    val userDto: PublicUserDto,
    val plainText: String,
    val html: String
)

File: webApp/src/api/Oldtypes/timeline/timelinePostDto.ts
import type { UserDto } from "../user/userDto"
import type { JSONContent } from "@tiptap/react";

export type TimeLinePostDto = {
  uuid: string;
  createdAt: LocalDateTime;
  updatedAt: LocalDateTime;
  deletedAt: LocalDateTime | null;
  content: JSONContent;
  userDto: UserDto;
  plainText: string;
  html: string;
}